io/net/virt-net/network_virtualization.py

```diff
@@ -594,6 +597,59 @@ def test_vnic_hmc_dlpar(self):
                 count += 1
         self.check_dmesg_error()
 
+    def test_vnic_eeh(self):
+        """
+        Perform EEH on vnic interface from vios
+        """
+        current_logport = self.get_active_device_logport(self.slot_num[0])
+        if not self.original_logport == current_logport:
+            self.trigger_failover(self.original_logport)
+        else:
+            self.log.info("Unable to set the logport to original one")
+        time.sleep(5)
+        self.session = Session(self.vios_ip, user=self.vios_user,
+                               password=self.vios_pwd)
+        self.session.cleanup_master()
+        if not wait.wait_for(self.session.connect, timeout=30):
+            self.fail("Failed connecting to VIOS")
+        cmd = "ioscli lsmap -all -vnic -cpid %s" % self.lpar_id
+        vnic_servers = self.session.cmd(cmd).stdout_text.splitlines()
+        device = self.find_device(self.mac_id[0])
+        temp_idx = vnic_servers.index("Client device name:" + device)
+        vnic_bankingdevice = vnic_servers[temp_idx - 3].split(":")[1]
+        s = pxssh.pxssh()
+        try:
+            s.login(self.vios_ip, self.vios_user, self.vios_pwd)
+        except Exception:
+            self.warn("Unable to login to vios")
+        time.sleep(2)
+        s.sendline("oem_setup_env")
+        time.sleep(5)
+        s.sendline("kdb")
+        time.sleep(5)
+        s.sendline("set scroll false")
+        time.sleep(5)
+        cmd = "mlxcent setacs %s" % vnic_bankingdevice
+        s.sendline(cmd)
+        time.sleep(2)
+        s.sendline("mlxcent pollq 0")
```

Comment:
all commands are specific to mellanox adapters, what if the backing adapter is non mellanox... please add those commands